### PR TITLE
Added more options to security.firewalls

### PR DESCRIPTION
**Pull request by K0nias**
Added "methods" and "hosts" options to security.firewalls

### CODE REVIEW

File: src/Silex/Provider/SecurityServiceProvider.php [hunk 1]
```diff
@@ -344,14 +371,14 @@ public function register(Container $app)
                 if (is_string($rule[0])) {
                     $rule[0] = new RequestMatcher($rule[0]);
                 } elseif (is_array($rule[0])) {
-                    $rule[0] += [
+                    $rule[0] += array(
```
Comment (emirb) [inline src/Silex/Provider/SecurityServiceProvider.php]: Please revert to short array syntax.

Comment (K0nias) [inline src/Silex/Provider/SecurityServiceProvider.php]: can you please tell me why? its more readable my way.

Comment (stof) [inline src/Silex/Provider/SecurityServiceProvider.php]: @emirb actually, all other code in Silex is using the long array syntax

Comment (emirb) [inline src/Silex/Provider/SecurityServiceProvider.php]: @stof My bad, as I presumed standard would be short array syntax as we're supporting versions only above 5.4.
@K0nias disregard my comment then :)

Comment (K0nias) [inline src/Silex/Provider/SecurityServiceProvider.php]: @emirb oh I misunderstand your comment. I thought that you were commenting associates array with string keys. Ok then. :)

**Comment by GromNaN** [comment]
👍 you should add a test case.

**Comment by K0nias** [comment]
@GromNaN added :)

**Comment by fabpot** [comment]
Thank you @K0nias.

**Pull request closed without merging**